plugins/rapid7_insight_agent/unit_test/test_get_agent_details.py

```diff
@@ -1,42 +1,54 @@
-import json
-import logging
+from icon_rapid7_insight_agent.actions.get_agent_details.schema import Input
 
 
+@patch("requests.sessions.Session.post", side_effect=Util.mocked_request)
 class TestGetAgentDetails(TestCase):
-    def test_integration_get_agent_details(self):
-        log = logging.getLogger("Test")
-        test_conn = Connection()
-        test_action = GetAgentDetails()
-
-        test_conn.logger = log
-        test_action.logger = log
-
-        try:
-            with open("../tests/get_agent_details.json") as file:
-                test_json = json.loads(file.read()).get("body")
-                connection_params = test_json.get("connection")
-                action_params = test_json.get("input")
-        except Exception as e:
-            message = """
-            Could not find or read sample tests from /tests directory
-            
-            An exception here likely means you didn't fill out your samples correctly in the /tests directory 
-            Please use 'icon-plugin generate samples', and fill out the resulting test files in the /tests directory
-            """
-            self.fail(message)
-
-        test_conn.connect(connection_params)
-        test_action.connection = test_conn
-        results = test_action.run(action_params)
-
-        self.assertTrue("agent" in results.keys())
-        self.assertTrue(len(results), 1)
-        self.assertEqual(results.get("agent").get("id"), "15eec979a15f75ad70567d58ad8a0aef")
+    @classmethod
+    @patch("requests.sessions.Session.post", side_effect=Util.mocked_request_for_api_key)
+    def setUpClass(cls, mock_request) -> None:
+        cls.action = Util.default_connector(GetAgentDetails())
+
+    def test_get_agent_by_hostname(self, mock_request):
+        actual = self.action.run({Input.AGENT: "ivm-agent-ub18"})
+        expect = Util.load_json("expected/get_agent_details.exp")
+        self.assertEqual(expect, actual)
+
+    def test_get_agent_by_hostname_with_none_in_response(self, mock_request):
+        actual = self.action.run({Input.AGENT: "ivm-agent-win"})
+        expect = Util.load_json("expected/get_agent_details_with_none.exp")
+        self.assertEqual(expect, actual)
+
+    def test_get_agent_by_ip(self, mock_request):
+        actual = self.action.run({Input.AGENT: "11.44.23.53"})
+        expect = Util.load_json("expected/get_agent_details.exp")
+        self.assertEqual(expect, actual)
+
+    def test_get_agent_by_ip_with_none_in_response(self, mock_request):
+        actual = self.action.run({Input.AGENT: "12.43.13.43"})
+        expect = Util.load_json("expected/get_agent_details_with_none.exp")
+        self.assertEqual(expect, actual)
+
+    def test_get_agent_by_mac(self, mock_request):
+        actual = self.action.run({Input.AGENT: "01-51-53-44-A8-D1"})
+        expect = Util.load_json("expected/get_agent_details.exp")
+        self.assertEqual(expect, actual)
```

Comment:
Could you please `parameterize` that tests? It's possible to put `Input.AGENT` and `expect` as a parameters - it's not needed to repeat the code 😃  Please use `parameterized` module to do such a thing.